.github/workflows/build-lint-test.yml

```diff
@@ -167,50 +135,6 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version-file: '.nvmrc'
-      - name: Restore Yarn dependencies
-        uses: actions/cache@v3
-        with:
-          path: ${{ needs.platform-compatibility-prepare.outputs.YARN_CACHE_DIR }}
-          key: yarn-cache-${{ runner.os }}-${{ needs.platform-compatibility-prepare.outputs.YARN_VERSION }}-${{ hashFiles('yarn.lock') }}
+          cache: yarn
       - run: yarn --immutable
       - run: yarn workspace @metamask/snaps-cli run test
-  check-workflows:
-    name: Check workflows
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: Download actionlint
-        id: download-actionlint
-        run: bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/7fdc9630cc360ea1a469eed64ac6d78caeda1234/scripts/download-actionlint.bash) 1.6.22
-        shell: bash
-      - name: Check workflow files
-        run: ${{ steps.download-actionlint.outputs.executable }} -color
-        shell: bash
-  all-jobs-completed:
-    name: All jobs completed
-    runs-on: ubuntu-latest
-    needs:
-      - build
-      - lint
-      - test
-      - platform-compatibility-test
-      - check-workflows
-    outputs:
-      PASSED: ${{ steps.set-output.outputs.PASSED }}
-    steps:
-      - name: Set PASSED output
-        id: set-output
-        run: echo "PASSED=true" >> "$GITHUB_OUTPUT"
-
-  all-jobs-pass:
-    name: All jobs pass
-    if: ${{ always() }}
```

Comment:
I noticed this was removed in `MetaMask/core`. Is this no longer required now?

@Gudahtt

Reply:
It was moved to `main.yml`; the basic strategy is still required, but no longer required here.

Reply:
I specifically meant the `if ${{ always() }}` part. It's not in the workflow in the core repo:

https://github.com/MetaMask/core/blob/0b3b7e96c6ab2aa636218f4a3fbf28814a6398f9/.github/workflows/main.yml#L50-L55

Reply:
Sorry I was referencing the module template, but you're right, that wouldn't be the right comparison for a monorepo. Good catch, I am not sure why this is missing. It seems to be working though, I see correctly-failing PRs.

Reply:
Scratch that; this is broken in the core monorepo. "All jobs passed" is showing as skipped when something fails, rather than as failed. So it's not acting as a required status check.

Reply:
I've updated the main workflow to make `all-jobs-pass` always run.